### PR TITLE
feat(crypto_box): allow for ed25519 conversion 

### PR DESCRIPTION
**Pull request by dignifiedquire**
Based on https://libsodium.gitbook.io/doc/advanced/ed25519-curve25519
Depends on https://github.com/dalek-cryptography/ed25519-dalek/pull/296

### CODE REVIEW

**Comment by tarcieri** [comment]
Perhaps this should go in `ed25519-dalek` rather than here?

**Comment by dignifiedquire** [comment]
So that ed25519 would depend on `crypto_box`?

File: Cargo.toml [hunk 1]
```diff
@@ -6,3 +6,6 @@ members = [
     "crypto_secretbox",
     "crypto_secretstream",
 ]
+
+[patch.crates-io]
+ed25519-dalek = { git = "https://github.com/dalek-cryptography/ed25519-dalek", branch = "scalar-and-montgomery-point-conversions"}
```
Comment (tarcieri) [inline Cargo.toml:11]: The PR has been merged.

```suggestion
ed25519-dalek = { git = "https://github.com/dalek-cryptography/ed25519-dalek" }
```

**Comment by tarcieri** [comment]
I think the only parts of this that are actually needed anymore are the `From` impls.

I think you can probably even drop `ed25519-dalek` from `dev-dependencies`, since that test is effectively testing the same thing as the test added in https://github.com/dalek-cryptography/ed25519-dalek/pull/296

**Comment by dignifiedquire** [comment]
> I think the only parts of this that are actually needed anymore are the `From` impls.

Done. Makes sense, now that there are tests in ed25519 for this logic.

**Pull request merged**